sahi/auto_model.py

```diff
@@ -119,16 +116,17 @@ def from_layer(
 
         layer_model = layer.get_model(name=model_path, no_cache=no_cache).get_train()
         if layer_model.__class__.__module__ in ["yolov5.models.common", "models.common"]:
-            model = Yolov5DetectionModel(
-                model=layer_model,
-                device=device,
-                mask_threshold=mask_threshold,
-                confidence_threshold=confidence_threshold,
-                category_mapping=category_mapping,
-                category_remapping=category_remapping,
-                image_size=image_size,
-            )
+            model_type = "yolov5"
         else:
             raise Exception(f"Unsupported model: {type(layer_model)}. Only YOLOv5 models are supported.")
 
-        return model
+        DetectionModel = MODEL_TYPE_TO_MODEL_CLASS_NAME[model_type]
+        return DetectionModel(
```

Comment:
MODEL_TYPE_TO_MODEL_CLASS_NAME is a `Dict[str, str]`, so `DetectionModel` is also a `str` object. I think we have a typo here, DetectionModel was first needs to be imported/loaded and then instantiated.

Reply:
fixed :+1: